### PR TITLE
Cache builder binary in Travis

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Fixes https://github.com/giantswarm/inago/issues/112

### CODE REVIEW

**Comment by zeisss** [comment]
There is no need for caching this folder. Actually it makes it worse when we change dependencies. `builder go get` already has a cache, so `~/.giantswarm/builder` should be cached IMO.

**Comment by JosephSalisbury** [comment]
I can't seem to get it to work anyways :D
I'm just gonna cache the actual `builder` binary.

**Comment by JosephSalisbury** [comment]
@giantswarm/cena RFR

File: .travis.yml [hunk 1]
```diff
@@ -4,13 +4,19 @@ language: go
...
+    directories:
+    - ./builder-cache
 
 notifications:
     email: false
 
 before_install:
-- wget https://downloads.giantswarm.io/builder/0.11.0/builder
-- chmod +x ./builder
+- cp ./builder-cache/builder-$BUILDER_VERSION ./builder || wget https://downloads.giantswarm.io/builder/$BUILDER_VERSION/builder && chmod +x ./builder
```
Comment (zeisss) [inline .travis.yml:19]: this feels to complex.

```
test -f ./builder-cache/builder-$BUILDER_VERSION || wget https://downloads.giantswarm.io/builder/$BUILDER_VERSION/builder -o ./builder-cache/builder-$BUILDER_VERSION
cp ./builder-cache/builder-$BUILDER_VERSION ./builder
chmod +x builder
```

this downlodas builder into the cache, if needed. No `before_cache` required.

Comment (zeisss) [inline .travis.yml:19]: actually not sure this is better, feel free to judge

**Comment by zeisss** [comment]
LGTM

File: .travis.yml (hunk 1 above)
Comment (JosephSalisbury) [inline .travis.yml:19]: can't say I feel much difference, and mine is smaller and ready to go :D

**Pull request merged**